docs/getting-started/server/guide-vscode.md

```diff
@@ -0,0 +1,270 @@
+---
+sidebar_position: 2
+---
+
+# VS Code Dev Containers
+
+This page will show you how to set up a local Bitwarden server for development purposes utilizing
+Visual Studio Code Dev Containers. You can read more about this in the
+[Visual Studio Code Dev Containers documentation](https://code.visualstudio.com/docs/devcontainers/containers).
+
+The Bitwarden server is comprised of several services that can run independently. For a basic
+development setup, you will need the **Admin**, **Api**, and **Identity** services.
+
+:::info
+
+Before you start: make sure you’ve installed the recommended
+[Tools and Libraries](../tools/index.md). Namely, the following:
+
+- Docker Desktop
+- [Visual Studio Code](https://code.visualstudio.com/)
+- Azure Data Studio
+
+:::
+
+## Clone the repository
+
+1.  Clone the Bitwarden Server project:
+
+    ```bash
+    git clone https://github.com/bitwarden/server.git
+    ```
+
+2.  Open a terminal and navigate to the root of the cloned repository.
+
+:::caution
+
+When first opening the repository in Visual Studio Code, you will be prompted to reopen the folder
+in a Dev Container. If you have not completed the steps in the
+[Configure Env File](#configure-env-file) section below, the creation of the Dev Container will
+fail.
+
+:::
+
+## Configure Git
+
+1. Configure Git to ignore the Prettier revision:
+
+   ```bash
+   git config blame.ignoreRevsFile .git-blame-ignore-revs
+   ```
+
+2. _(Optional)_ Set up the pre-commit `dotnet format` hook:
+
+   ```bash
+   git config --local core.hooksPath .git-hooks
+   ```
+
+   Formatting requires a full build, which may be too slow to do every commit. As an alternative,
+   you can run `dotnet format` from the command line when convenient (e.g. before requesting a PR
+   review).
+
+## Configure Env File
+
+The Dev Container will automatically load the environment variables from the values in `dev/.env`.
+To get started, copy the example file:
+
+1. ```bash
+   cd dev
+   cp .env.example .env
+   ```
+
+2. Open `.env` with your preferred editor.
+
+3. Set the `MSSQL_SA_PASSWORD` variable. This will be the password for your MSSQL database server.
+
+:::caution
+
+Your MSSQL password must comply with the following
+[password complexity guidelines](https://docs.microsoft.com/en-us/sql/relational-databases/security/password-policy?view=sql-server-ver15#password-complexity)
+
+- It must be at least eight characters long.
+- It must contain characters from three of the following four categories:
+- Latin uppercase letters (A through Z)
+- Latin lowercase letters (a through z)
+- Base 10 digits (0 through 9)
+- Non-alphanumeric characters such as: exclamation point (!), dollar sign ($), number sign (#), or
+  percent (%).
+
+:::
+
+4.  You can change the other variables or use their default values. Save and quit this file.
+
+### SQL Server
```

Comment:
The first thought that comes to mind if we wanted to have slight variance in the steps that _do_ still manually require some user interaction would be to add a "Dev Container" option to the top-right dropdown menu, or something like that.

Doing so might also be weird though because that menu currently distinguishes between internal engineering and external community dev configuration guides.